src/Microsoft.OData.Core/JsonLight/ODataJsonLightReaderNestedResourceInfo.cs

```diff
@@ -65,6 +61,7 @@ private ODataJsonLightReaderNestedResourceInfo(ODataNestedResourceInfo nestedRes
 
             this.nestedResourceInfo = nestedResourceInfo;
             this.nestedProperty = nestedProperty;
+            NestedResourceTypeReference = (nestedProperty != null) ? nestedProperty.Type.ToStructuredType().ToTypeReference(nestedProperty.Type.IsNullable).AsStructured() : null;

```

Comment:
>nestedProperty.Type.ToStructuredType().ToTypeReference(nestedProperty.Type.IsNullable).AsStructured() [](start = 69, length = 101)

You can just use "nestedProperty.Type..AsStructured()" directly

Reply:
I need to use ToStructuredType to correctly handle collections. I have cleaned up the code little though. 

---
In reply to: [267146833](https://github.com/OData/odata.net/pull/1322#discussion_r267146833) [](ancestors = 267146833)